File: server/internal/model/message.go
package model

import (
	"time"
)

// MessageType 消息类型
type MessageType string

const (
	MessageTypeText  MessageType = "text"
	MessageTypeImage MessageType = "image"
	MessageTypeFile  MessageType = "file"
	MessageTypeAudio MessageType = "audio"
	MessageTypeVideo MessageType = "video"
)

// Message 消息模型
type Message struct {
	ID             uint        `json:"id" gorm:"primaryKey"`
	ConversationID uint        `json:"conversation_id" gorm:"index;not null"`
	SenderID       uint        `json:"sender_id" gorm:"index;not null"`
	Type           MessageType `json:"type" gorm:"default:'text'"`
	Content        string      `json:"content" gorm:"type:text"`
	MediaURL       string      `json:"media_url,omitempty"` // 图片/文件/音视频URL
	IsRead         bool        `json:"is_read" gorm:"default:false"`
	IsDeleted      bool        `json:"is_deleted" gorm:"default:false"`
	CreatedAt      time.Time   `json:"created_at"`
	UpdatedAt      time.Time   `json:"updated_at"`

	// 关联
	Sender *User `json:"sender,omitempty" gorm:"foreignKey:SenderID"`
}

// TableName 指定表名
func (Message) TableName() string {
	return "messages"
}

// Conversation 会话模型
type Conversation struct {
	ID            uint      `json:"id" gorm:"primaryKey"`
	Type          string    `json:"type" gorm:"default:'private'"` // private, group
	Name          string    `json:"name"`                          // 群聊名称
	Avatar        string    `json:"avatar"`                        // 群聊头像
	LastMessageID uint      `json:"last_message_id"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`

	// 关联
	LastMessage  *Message               `json:"last_message,omitempty" gorm:"foreignKey:LastMessageID"`
	Participants []*ConversationMember  `json:"participants,omitempty" gorm:"foreignKey:ConversationID"`
}

// TableName 指定表名
func (Conversation) TableName() string {
	return "conversations"
}

// ConversationMember 会话成员
type ConversationMember struct {
	ID             uint      `json:"id" gorm:"primaryKey"`
	ConversationID uint      `json:"conversation_id" gorm:"index;not null"`
	UserID         uint      `json:"user_id" gorm:"index;not null"`
	UnreadCount    int       `json:"unread_count" gorm:"default:0"`
	JoinedAt       time.Time `json:"joined_at"`
	LastReadAt     time.Time `json:"last_read_at"`

	// 关联
	User *User `json:"user,omitempty" gorm:"foreignKey:UserID"`
}

// TableName 指定表名
func (ConversationMember) TableName() string {
	return "conversation_members"
}

// SendMessageRequest 发送消息请求
type SendMessageRequest struct {
	ConversationID uint        `json:"conversation_id" binding:"required"`
	Type           MessageType `json:"type" binding:"required"`
	Content        string      `json:"content"`
	MediaURL       string      `json:"media_url,omitempty"`
}

// MessageResponse 消息响应
type MessageResponse struct {
	ID             uint         `json:"id"`
	ConversationID uint         `json:"conversation_id"`
	SenderID       uint         `json:"sender_id"`
	Type           MessageType  `json:"type"`
	Content        string       `json:"content"`
	MediaURL       string       `json:"media_url,omitempty"`
	IsRead         bool         `json:"is_read"`
	CreatedAt      time.Time    `json:"created_at"`
	Sender         *UserResponse `json:"sender,omitempty"`
}

// ToResponse 转换为响应格式
func (m *Message) ToResponse() *MessageResponse {
	resp := &MessageResponse{
		ID:             m.ID,
		ConversationID: m.ConversationID,
		SenderID:       m.SenderID,
		Type:           m.Type,
		Content:        m.Content,
		MediaURL:       m.MediaURL,
		IsRead:         m.IsRead,
		CreatedAt:      m.CreatedAt,
	}

	if m.Sender != nil {
		resp.Sender = m.Sender.ToResponse()
	}

	return resp
}

File: client/ChatApp/Models/Message.swift
import Foundation

/// 消息类型
enum MessageType: String, Codable {
    case text
    case image
    case file
    case audio
    case video
}

/// 消息模型
struct Message: Codable, Identifiable, Equatable {
    let id: UInt
    let conversationId: UInt
    let senderId: UInt
    let type: MessageType
    let content: String
    let mediaUrl: String?
    let isRead: Bool
    let createdAt: Date
    let sender: User?

    enum CodingKeys: String, CodingKey {
        case id
        case conversationId = "conversation_id"
        case senderId = "sender_id"
        case type
        case content
        case mediaUrl = "media_url"
        case isRead = "is_read"
        case createdAt = "created_at"
        case sender
    }

    /// 是否是自己发送的消息
    func isMe(userId: UInt) -> Bool {
        senderId == userId
    }

    /// 格式化时间显示
    var timeString: String {
        let formatter = DateFormatter()
        let calendar = Calendar.current

        if calendar.isDateInToday(createdAt) {
            formatter.dateFormat = "HH:mm"
        } else if calendar.isDateInYesterday(createdAt) {
            return "Yesterday"
        } else {
            formatter.dateFormat = "MM/dd"
        }

        return formatter.string(from: createdAt)
    }
}

/// 发送消息请求
struct SendMessageRequest: Codable {
    let conversationId: UInt
    let type: MessageType
    let content: String
    let mediaUrl: String?

    enum CodingKeys: String, CodingKey {
        case conversationId = "conversation_id"
        case type
        case content
        case mediaUrl = "media_url"
    }
}

/// 消息响应
struct MessageResponse: Codable {
    let message: String
    let data: Message
}

/// 获取消息列表响应
struct MessagesResponse: Codable {
    let messages: [Message]
    let page: Int
    let pageSize: Int

    enum CodingKeys: String, CodingKey {
        case messages
        case page
        case pageSize = "page_size"
    }
}
